### ISSUE TITLE
pe:exporter - Improve handling of HtmlOutputLink

### ISSUE THREAD

**Issue by andrei-ivanov** [primefaces-extensions/primefaces-extensions.github.com#286]
`org.primefaces.extensions.component.exporter.Exporter.exportValue` should handle instances of HtmlOutputLink and export the label of the link, not the value, which is the URL (or maybe even create a link where it makes sense, like for Excel).
```java
if (component instanceof HtmlOutputLink) {
	//export first value holder
	for (UIComponent child : component.getChildren()) {
		if (child instanceof ValueHolder) {
			return exportValue(context, child);
		}
	}

	return "";
}
```

**Comment by tandraschko** [comment]
would you please check this sudheer? If the patch looks correct, please apply it.

**Comment by andrei-ivanov** [comment]
My workaround:
```java
import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
@Aspect
public class ExcelExporterEnhancingAspect {
	@Around("execution(* org.primefaces.extensions.component.exporter.Exporter.exportValue(..)) && args(context, component)")
	public Object exportValue(ProceedingJoinPoint call, FacesContext context, UIComponent component) throws Throwable {
		if (component instanceof HtmlOutputLink) {
			//export first value holder
			for (UIComponent child : component.getChildren()) {
				if (child instanceof ValueHolder) {
					return call.proceed(new Object[] { context, child });
				}
			}

			return "";
		}
		return call.proceed();
	}
}
```
Aspect applied using load time weaving.

**Comment by sudheerj** [comment]
The above code snippets still exports the link value but not display value.

**Comment by andrei-ivanov** [comment]
Hmm, my workaround, which does the same thing as the snippet in the issue description, works fine in production for more than 1 and a half years.

Can you show why it doesn't work?

**Comment by sudheerj** [comment]
The above code snippets makes value/link to get exported but we need to export the label.We fixed the issue in 6.0 release.Thanks.

**Comment by andrei-ivanov** [comment]
If you read again the description of the ticket you will see that this is exactly what I requested, to export the label and not the URL.
Can you tell me what is the commit that fixes the issue?
